### PR TITLE
Add tests for schedulercache

### PR DESCRIPTION
**Pull request by huangjiuyuan**
**What this PR does / why we need it**:
Add tests for `node_info.go` under `schedulercache` package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

**Comment by bsalamat** [comment]
@aveshagarwal Could you please take a look at this PR? Thanks!

**Comment by huangjiuyuan** [comment]
ping @aveshagarwal

**Comment by aveshagarwal** [comment]
@huangjiuyuan I will take a look tomorrow.

**Comment by aveshagarwal** [comment]
Added sig scheduling label otherwise its was difficult to find this PR.

File: pkg/scheduler/schedulercache/node_info_test.go [hunk 1]
```diff
@@ -0,0 +1,850 @@
...
+
+	for _, test := range tests {
+		r := NewResource(test.resourceList)
+		if !reflect.DeepEqual(test.expected, r) {
+			t.Errorf("expected: %#v, got: %#v", test.expected, r)
+		}
+	}
+}
+
+func TestResourceList(t *testing.T) {
```
Comment (aveshagarwal) [inline pkg/scheduler/schedulercache/node_info_test.go]: I dont think this test is really required here.

Comment (aveshagarwal) [reply pkg/scheduler/schedulercache/node_info_test.go]: nevermind, you can keep it.

File: pkg/scheduler/schedulercache/node_info_test.go [hunk 2]
```diff
@@ -0,0 +1,850 @@
...
+		ni := test.nodeInfo.Clone()
+		// Modify the field to check if the result is a clone of the origin one.
+		test.nodeInfo.generation += 10
+		if !reflect.DeepEqual(test.expected, ni) {
+			t.Errorf("expected: %#v, got: %#v", test.expected, ni)
+		}
+	}
+}
+
+func TestNodeInfoAddPod(t *testing.T) {
```
Comment (aveshagarwal) [inline pkg/scheduler/schedulercache/node_info_test.go]: This is not very different to TestNewNodeInfo as NewNodeInfo calls AddPod. May be you could just merge these.

Comment (aveshagarwal) [reply pkg/scheduler/schedulercache/node_info_test.go]: Or start with a node that already has some pods, and then add pod to the node.

File: pkg/scheduler/schedulercache/node_info_test.go [hunk 3]
```diff
@@ -0,0 +1,850 @@
...
+							NodeName: nodeName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		ni := fakeNodeInfo()
```
Comment (aveshagarwal) [inline pkg/scheduler/schedulercache/node_info_test.go]: 1. start with a node that has some pods already.

Comment (aveshagarwal) [reply pkg/scheduler/schedulercache/node_info_test.go]: or call addnode more than one on an empty node.

**Comment by aveshagarwal** [comment]
@huangjiuyuan thanks for this PR, just some minor comments, otherwise lgtm.

**Comment by huangjiuyuan** [comment]
@aveshagarwal Thanks for your kindness. I have changed a little bit to add more than one pods to an empty node.

**Comment by bsalamat** [comment]
/approve

Thanks, @huangjiuyuan and @aveshagarwal!

**Pull request merged**